src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2360,6 +2360,59 @@ void *instanceservicesession(void *dummyPt) {
                 JasmineGraphInstance::sendFileThroughService(orgHost, stoi(orgDataPort), filename, filepath);
             }
 
+        }
+        else if (line.compare(JasmineGraphInstanceProtocol::DISTRUBUTE_SHARED_CLUSTERS) == 0) {
+            instance_logger.log("Received : " + JasmineGraphInstanceProtocol::DISTRUBUTE_SHARED_CLUSTERS, "info");
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+
+            //Read requested graph from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string graphID = (data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
+
+            //Read org ID from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string orgID = (data);
+            graphID = utils.trim_copy(orgID, " \f\n\r\t\v");
+            instance_logger.log("Received org ID: " + orgID, "info");
+
+            //Read host partition details from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string hostPartitionListStr = (data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + hostPartitionListStr, "info");
+
+            //Parse host cluster details string
+            vector<string> partitionLocales = utils.split(hostPartitionListStr, '|');
+            //Iterate through each worker partition pair and collect from other workers
+            int count = 0;
+            std::thread *workerThreads = new std::thread[partitionLocales.size()];
+            for (auto p: partitionLocales) {
+                vector<string> workerPartition = utils.split(p, ',');
+                string hostname = workerPartition.at(0);
+                int port = stoi(workerPartition.at(1));
+                int dataPort = stoi(workerPartition.at(2));
+                string cluster = workerPartition.at(3);
+
+                string filename = "attrcluster_" + orgID + "_" + graphID + cluster + ".txt";
+                string filepath =
+                        utils.getJasmineGraphProperty("org.jasminegraph.server.instance.entityresolutionfolder") + filename;
```

Comment:
In this part of the code,

```
tils.getJasmineGraphProperty("org.jasminegraph.server.instance.entityresolutionfolder") + filename
```

we assume that there is a separator which user have given at the end of the `org.jasminegraph.server.instance.entityresolutionfolder` property. However, this may not be the case. We need to have a `/` added between the property and the filename so that we can avoid issues due to not having the `/`.